builtin/logical/transit/path_random.go

```diff
@@ -74,9 +97,39 @@ func (b *backend) pathRandomWrite(ctx context.Context, req *logical.Request, d *
 		return logical.ErrorResponse("unsupported encoding format %q; must be \"hex\" or \"base64\"", format), nil
 	}
 
-	randBytes, err := uuid.GenerateRandomBytes(bytes)
-	if err != nil {
-		return nil, err
+	var randBytes []byte
+	var warning string
+	switch source {
+	case "", "platform":
+		randBytes, err = uuid.GenerateRandomBytes(bytes)
+		if err != nil {
+			return nil, err
+		}
+	case "seal":
+		if rand.Reader == b.GetRandomReader() {
+			warning = "no seal/entropy augmentation available, using platform entropy source"
+		}
+		randBytes = make([]byte, bytes)
+		_, err = io.ReadFull(b.GetRandomReader(), randBytes)
```

Comment:
There is `uuid.GenerateRandomBytesWithReader`, which handles this for you if you want.